src/rule_dsl.jl

```diff
@@ -233,22 +233,42 @@ struct RuleRewriteError
     expr
 end
 
-function (r::RuleSet)(term; depth=typemax(Int), applyall=false, recurse=true)
+node_count(atom, count; cutoff) = count + 1
+node_count(t::Term, count=0; cutoff=100) = sum(node_count(arg, count; cutoff=cutoff) for arg ∈ arguments(t))
+
+function _recurse_apply_ruleset(r::RuleSet, term; depth, recurse, applyall, threaded=false, thread_depth_cutoff=100)
+    kwargs = (;applyall=applyall, recurse=recurse, threaded=threaded,
+              thread_depth_cutoff=thread_depth_cutoff)
+    if threaded
+        _args = map(arguments(term)) do arg
+            if node_count(term) > thread_depth_cutoff
```

Comment:
should it be called `thread_subtree_size` or `thread_descendent_cutoff`. depth usually means no. of levels.